### ISSUE TITLE
Unable to add firewall rules to IP Tables: Nomad enforces first rule

### ISSUE THREAD

**Issue by microadam** [hashicorp/nomad#10180]
### Nomad version
Nomad v1.0.4 (9294f35f9aa8dbb4acb6e85fa88e3e2534a3e41a)

### Issue

I am unable to add any additional firewall rules to the `NOMAD-ADMIN` IPTables chain, as nomad enforces that its ACCEPT rule is the first rule. So any rules that I append before nomad runs, are skipped.

This seems to be the offending line:
https://github.com/hashicorp/nomad/blob/a82697a2f024555cd68d0fa4ba5b52257d885e6b/client/allocrunner/networking_bridge_linux.go#L78

Is there any reason to ensure it's always the first rule? 

According to https://www.cni.dev/plugins/meta/firewall/

The admin chain `is intended as an user-controlled chain for custom rules that run prior to rules managed by the firewall plugin`

### Reproduction steps

Add an IP Tables rule before nomad runs to the `NOMAD-ADMIN` chain. Run nomad with a container that exposes a port.

#### Expected Result

Nomad to append its rule, so any existing rules are executed first.

```
-A NOMAD-ADMIN -j DOCKER-USER >>>>>>> CUSTOM RULE
-A NOMAD-ADMIN -d 172.26.64.0/20 -o nomad -j ACCEPT >>>>>> NOMAD RULE
```

#### Actual Result

Rule is ignored as nomad enforces its rule to be first
```
-A NOMAD-ADMIN -d 172.26.64.0/20 -o nomad -j ACCEPT >>>>>> NOMAD RULE
-A NOMAD-ADMIN -j DOCKER-USER >>>>>>> CUSTOM RULE
```

**Comment by tgross** [comment]
Looks like that link needs to be https://www.cni.dev/plugins/current/meta/firewall/#forward:

> The `CNI-FORWARD` chain first sends all traffic to `CNI-ADMIN` chain, which is intended as an user-controlled chain for custom rules that run prior to rules managed by the `firewall` plugin. The firewall plugin does not add, delete or modify rules in the `CNI-ADMIN` chain.

This seems reasonable to me. The existing behavior was added in https://github.com/hashicorp/nomad/commit/99742f2665a4b1470734cf9d88049f2ad32a2445 and the doc comment here from @nickethier is awfully suggestive that I might be missing something as to why that could be a problem. So tagging him and @shoenig for a consult here.

**Comment by microadam** [comment]
Thanks for taking a look! I believe the linked PR that I made is completely backwards compatible, as the `NOMAD-ADMIN` chain is purely controlled by Nomad, it only adds one rule to it. So if others don't care about adding firewall rules, it would still be the first and only rule in the chain if its appended (as opposed to being forced to be first). Would very much love to know if there is a good reason to have it as always first, as it seems like quite a conscious design decision. Any input from @nickethier  would def be most appreciated :)

**Comment by microadam** [comment]
@tgross @nickethier Not sure what the usual time frame for responding is, so apologies if I am prematurely following up and there is a process in place / queue that I am not aware of! Would just love to know the if the associated PR is something that could be merged, or if I need to find an alternative solution / workaround

**Comment by tgross** [comment]
No worries about pinging us, especially after I totally said we'd look at it. I wanted to do some testing to make sure we understood what would happen around Nomad client restarts and it looks ok. Going to take one more pass over it in the next day or so and then merge if there's nothing we run into.

**Comment by microadam** [comment]
Thank you very much for the quick response! Sounds excellent to me :)

**Comment by tgross** [comment]
#10181 has been merged and will ship in the upcoming Nomad 1.1.0
